asf_core_data/pipeline/data_joining/merge_proc_datasets.py

```diff
@@ -0,0 +1,206 @@
+# File: asf_core_data/pipeline/data_joining/merge_proc_datasets.py
+"""Join the EPC and MCS datasets."""
+
+# ---------------------------------------------------------------------------------
+
+from asf_core_data.getters import data_getters
+from asf_core_data.config import base_config
+from asf_core_data.getters.epc import data_batches
+from asf_core_data.pipeline.data_joining import merge_install_dates
+from asf_core_data import load_preprocessed_epc_data
+from asf_core_data.pipeline.preprocessing import data_cleaning
+
+import pandas as pd
+import numpy as np
+
+# ---------------------------------------------------------------------------------
+
+
+def merge_proc_epc_and_mcs_installations(epc_df, verbose=False):
+    """Merge processed EPC and MCS installations data.
+
+    - Load MCS installations data (most relevant fields)
+    - Mark matches in data (MCS_AVAILABLE, EPC_AVAILABLE)
+    - Join based on UPRN, otherwise concatenate
+    - Standardise fields such as HP_INSTALLED or HP_TYPE
+
+    Args:
+        epc_df (pd.DataFrame): Processed EPC dataframe.
+        verbose (bool, optional): Print shape of dataframes as they are merged.
+
+    Returns:
+        pd.DataFrame: Merged EPC and MCS installations dataframe.
+    """
+
+    newest_joined_batch = data_batches.get_latest_mcs_epc_joined_batch()
+
+    # # Load MCS
+    mcs_df = data_getters.load_s3_data(
+        base_config.BUCKET_NAME,
+        newest_joined_batch,
+        # TODO: Check whether we need any other fields
+        usecols=[
+            "UPRN",
+            "commission_date",
+            "capacity",
+            "estimated_annual_generation",
+            "flow_temp",
+            "tech_type",
+            "scop",
+            "design",
+            "product_name",
+            "manufacturer",
+            "cost",
+        ],
+        dtype={"UPRN": "str", "commission_date": "str"},
+    )
+
+    # Get the MCS install dates (still necessary?)
+    mcs_df["commission_date"] = pd.to_datetime(
+        mcs_df["commission_date"]
+        .str.strip()
+        .str.lower()
+        .replace(r"\s.*", "", regex=True)
+        .replace(r"-", "", regex=True),
+        format="%Y%m%d",
+    )
+
+    mcs_df.rename(columns={"postcode": "POSTCODE"}, inplace=True)
+
+    # Tag whether MCS or EPC match is available
+    mcs_df["MCS_AVAILABLE"] = True
+    epc_df["EPC_AVAILABLE"] = True
```

Comment:
is this line necessary?

Reply:
Yes, for tracking whether EPC or MCS is available for a record, we have to set the defaults to True. Then further down we change to False for the ones with missing connections between the two sets.